pkg/updatestatus/controlplaneinformer.go

```diff
@@ -23,72 +26,258 @@ import (
 	}
 
-	now := c.now()
-	insight := assessClusterVersion(clusterVersion, now)
-	msg := makeInsightMsgForClusterVersion(insight, now)
+	var msg informerMsg
+	switch t {
+	case clusterVersionKindName:
+		clusterVersion, err := c.clusterVersions.Get(name)
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				// TODO: Handle deletes by deleting the status insight
+				return nil
+			}
+			return err
+		}
+
+		now := c.now()
+		insight := assessClusterVersion(clusterVersion, now)
+		msg = makeInsightMsgForClusterVersion(insight, now)
+
+	case clusterOperatorKindName:
+		clusterVersion, err := c.clusterVersions.Get("version")
+		if err != nil {
+			return err
+		}
+		targetVersion := clusterVersion.Status.Desired.Version
+
+		clusterOperator, err := c.clusterOperators.Get(name)
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				// TODO: Handle deletes by deleting the status insight
+				return nil
+			}
+			return err
+		}
+
+		now := c.now()
+		insight, err := assessClusterOperator(ctx, clusterOperator, targetVersion, c.appsClient, now)
+		if err != nil {
+			return fmt.Errorf("failed to assess cluster operator %s: %w", name, err)
+		}
+		msg = makeInsightMsgForClusterOperator(insight, now)
+	default:
+		return fmt.Errorf("invalid queue key %s with unexpected type %s", queueKey, t)
+	}
 	var msgForLog string
 	if klog.V(4).Enabled() {
 		msgForLog = fmt.Sprintf(" | msg=%s", string(msg.insight))
 	}
-	klog.V(2).Infof("CPI :: Syncing ClusterVersion %s%s", clusterVersion.Name, msgForLog)
+	klog.V(2).Infof("CPI :: Syncing %s %s%s", t, name, msgForLog)
 	c.sendInsight(msg)
 
 	return nil
 }
 
+func makeInsightMsgForClusterOperator(coInsight *ClusterOperatorStatusInsight, acquiredAt metav1.Time) informerMsg {
+	uid := fmt.Sprintf("usc-co-%s", coInsight.Name)
+	insight := Insight{
+		UID:        uid,
+		AcquiredAt: acquiredAt,
+		InsightUnion: InsightUnion{
+			Type:                         ClusterOperatorStatusInsightType,
+			ClusterOperatorStatusInsight: coInsight,
+		},
+	}
+	// Should handle errors, but ultimately we will have a proper API and won’t need to serialize ourselves
+	rawInsight, _ := yaml.Marshal(insight)
+	return informerMsg{
+		uid:     uid,
+		insight: rawInsight,
+	}
+}
+
+func assessClusterOperator(ctx context.Context, operator *configv1.ClusterOperator, targetVersion string, appsClient appsv1client.AppsV1Interface, now metav1.Time) (*ClusterOperatorStatusInsight, error) {
+	updating := metav1.Condition{
+		Type:               string(ClusterOperatorStatusInsightUpdating),
+		Status:             metav1.ConditionUnknown,
+		Reason:             string(ClusterOperatorUpdatingCannotDetermine),
+		LastTransitionTime: now,
+	}
+
+	imagePullSpec, err := getImagePullSpec(ctx, operator.Name, appsClient)
+	if err != nil && !errors.Is(err, operatorImageNotImplemented) {
+		return nil, err
+	}
+	if imagePullSpec == "" {
+		klog.Warningf("The image pull spec for %s is the empty string", operator.Name)
```

Comment:
No image pull spec is standard for most COs now, and we will be assessing many CO events, so logging a warning every time we do is too noisy. I don't think we need to log this at all? For MCO we'll get an error that makes us return on L170, and for other operators no imagePullSpec is normal.